FreeRTOS/FreeRTOS
# Adding unit tests for stream buffer and message buffer

Description
-----------

* Added CMock unit tests for stream buffer and message buffer
* Coverage  - 100% line, branch and functional coverage
* Update kernel submodule pointer to latest main to get the fix for stream buffer failing test.

Test Steps
-----------
Run only stream buffer tests:
`make stream_buffer`

Run only message buffer tests:
`make message_buffer`

Run whole suite with coverage:
`make`

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
